### PR TITLE
feat(angular) : Adding  text component in angular primitive poc.

### PR DESCRIPTION
**Pull request by SantoshLatpate**
#### Description of changes
1. Changes related to text components.
2. Added example component of text component.

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: examples/angular/src/pages/ui/primitive-poc/text/aws-exports.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+import awsExports from '@environments/auth-with-username-no-attributes/src/aws-exports';
+export default awsExports;
```
[the remark this answers is not in the document]
Comment (SantoshLatpate) [reply examples/angular/src/pages/ui/primitive-poc/text/aws-exports.js]: Removed aws-export.js

File: examples/angular/src/pages/ui/primitive-poc/text/text.component.html [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<p
+  amplify-text-poc
+  variation="primary"
+  color="blue"
+  lineHeight="1.5em"
+  fontWeight="{400}"
```
[the remark this answers is not in the document]
Comment (SantoshLatpate) [reply examples/angular/src/pages/ui/primitive-poc/text/text.component.html]: Done.

File: examples/angular/src/pages/ui/primitive-poc/text/text.component.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { Component, Input } from '@angular/core';
+
+@Component({
+  selector: 'amplify-view-demo',
+  templateUrl: 'text.component.html',
+})
+export class AmplifyTextExampleComponent {
+  @Input() buttonName = 'Amplify text POC ';
+  @Input() flag = true;
```
[the remark this answers is not in the document]
Comment (SantoshLatpate) [reply examples/angular/src/pages/ui/primitive-poc/text/text.component.ts]: Removed.

File: packages/angular/projects/ui-angular/src/lib/primitives-poc/text/text-component/text.component.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+@Component({
+  selector: $COMPONENT_SELECTOR,
+  template: '<ng-content></ng-content>',
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class AmplifyTextComponent extends AmplifyBasePrimitiveComponent {
+  /** Input Prop for style changes  */
+
+  @Input() variation: boolean = false;
```
[the remark this answers is not in the document]
Comment (SantoshLatpate) [reply packages/angular/projects/ui-angular/src/lib/primitives-poc/text/text-component/text.component.ts]: Done.

File: examples/angular/src/pages/ui/primitive-poc/text/text.component.html [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+  fontSize="1em"
+  fontStyle="normal"
+  textDecoration="none"
+  width="30vw"
+>
+  Hello World!
+</p>
+<br />
```
[the remark this answers is not in the document]
Comment (SantoshLatpate) [reply examples/angular/src/pages/ui/primitive-poc/text/text.component.html:106]: Done.

File: packages/angular/projects/ui-angular/src/lib/primitives-poc/text/text-component/text.component.ts [hunk 6]
```diff
@@ -14,14 +14,15 @@ const $COMPONENT_SELECTOR: string = '[amplify-text-poc]';
   encapsulation: ViewEncapsulation.None,
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
+
+/** Base primitive for property binding */
 export class AmplifyTextComponent extends AmplifyBasePrimitiveComponent {
   /** Input Prop for style changes  */
 
-  @Input() variation: boolean = false;
+  @Input() variation: string = 'primary';
```
Comment (wlee221) [inline packages/angular/projects/ui-angular/src/lib/primitives-poc/text/text-component/text.component.ts:22]: Can we strictly type this?

```ts
type TextVariation = 'primary' | 'secondary' | 'tertiary' | 'error' | 'warning' | 'info' | 'success'
```

File: examples/angular/src/pages/ui/primitive-poc/text/text.component.html (hunk 5 above)
[the remark this answers is not in the document]
Comment (SantoshLatpate) [reply examples/angular/src/pages/ui/primitive-poc/text/text.component.html:106]: https://github.com/aws-amplify/amplify-ui/commit/dbae75766d6daaa595443228d5faea3849102cb7

File: packages/angular/projects/ui-angular/src/lib/primitives-poc/text/text-component/text.component.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (SantoshLatpate) [reply packages/angular/projects/ui-angular/src/lib/primitives-poc/text/text-component/text.component.ts]: https://github.com/aws-amplify/amplify-ui/commit/dbae75766d6daaa595443228d5faea3849102cb7

File: examples/angular/src/pages/ui/primitive-poc/text/text.component.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (SantoshLatpate) [reply examples/angular/src/pages/ui/primitive-poc/text/text.component.ts]: https://github.com/aws-amplify/amplify-ui/commit/dbae75766d6daaa595443228d5faea3849102cb7

File: examples/angular/src/pages/ui/primitive-poc/text/text.component.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (SantoshLatpate) [reply examples/angular/src/pages/ui/primitive-poc/text/text.component.html]: https://github.com/aws-amplify/amplify-ui/commit/dbae75766d6daaa595443228d5faea3849102cb7

File: examples/angular/src/pages/ui/primitive-poc/text/aws-exports.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (SantoshLatpate) [reply examples/angular/src/pages/ui/primitive-poc/text/aws-exports.js]: https://github.com/aws-amplify/amplify-ui/commit/dbae75766d6daaa595443228d5faea3849102cb7

**Review by wlee221 — APPROVED** [review]

File: packages/angular/projects/ui-angular/src/lib/primitives-poc/text/text-component/text.component.ts (hunk 6 above)
Comment (SantoshLatpate) [reply packages/angular/projects/ui-angular/src/lib/primitives-poc/text/text-component/text.component.ts]: https://github.com/aws-amplify/amplify-ui/pull/2415/commits/f3d9e3121d464ab6cdf223cbf3f9967bce3ac4dc

**Pull request merged**